magento/magento2
# Consumers -  empty string prefix

### Description (*)
When using the DB for locks, consumers will not run if the lock prefix is set to '' (empty string) in env.php. It seems like the code only accounts for the prefix not being set at all (null). However, when running `setup:config:set`, the prefix value will again be set to be an empty string. 

```
    'lock' => [
        'provider' => 'db',
        'config' => [
            'prefix' => ''
        ]
    ],
```

Consumers run fine if any prefix is choosen, but not when prefix is empty. 



### Related Pull Requests
NIL

### Fixed Issues (if relevant)

NIL

### Manual testing scenarios (*)

1. Use DB as lock provider, and setup prefix to be an empty string
2. Run bulk actions, Exports etc


There seems to be a related issue whith the behavior of setup:config:set, that makes it even more important that consumers work with both null and an empty string. 

1. Remove the config-> prefix so you have the following in your `env.php`
```
    'lock' => [
        'provider' => 'db'
    ],
```
2. Run any valid `setup:config:set` command
3. See the change to `env.php`
```
    'lock' => [
        'provider' => 'db',
        'config' => [
            'prefix' => null
        ]
    ],
```
4. Run any valid `setup:config:set` command
5. See the change to `env.php`
```
    'lock' => [
        'provider' => 'db',
        'config' => [
            'prefix' => ''
        ]
    ],
```

If the prefix is not set, `setup:config:set` will change the value to `null`, and if the value is `null`, `setup:config:set` will change the value to an empty string. 

### Questions or comments
It was difficult to reproduce the issue on existing shops where consumer was allready running. However we can reproduce the issue consistently when we setup a new shop on a new server. 
I suspect the reason for the inconsistency when testing on an existing shop,  could be because a consumer was already spawned when we tested the changes, and the consumer only reads the settings upon creation and not when running (which would make sense)

### Contribution checklist (*)
